### PR TITLE
Build with Java 17 as well

### PR DESCRIPTION
**Pull request by klaraward**

### CODE REVIEW

**Comment by klaraward** [comment]
Need to make sure tests also run with the JVM args

**Comment by herder** [comment]
Is this an interim solution until we have a newer version of the format plugin?

File: pom.xml [hunk 1]
```diff
@@ -210,6 +210,14 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>jdk17+</id>
+            <activation>
+                <jdk>[17,)</jdk>
+            </activation>
+            <properties>
```
Comment (herder) [inline pom.xml:218]: Do we need both this profile and the `jvm.config` file? I'd prefer this one if not.

Also, we could add a todo to remove it once we have upgraded google-fmt to a version that supports 17:

```suggestion
            <properties>
            <!-- TODO: Remove this once google-java-fmt supports 17 -->
```

**Review by herder — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (klaraward) [reply pom.xml:218]: One is for the tests, one is for the actual plugin that formats code in this repo.

**Review by caesar-ralf — APPROVED** [review]

**Review by herder — APPROVED** [review]

**Pull request merged**